libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -331,6 +332,45 @@ def from_texts(
         """Return VectorStore initialized from texts and embeddings."""
         return super().from_texts(texts, embedding, metadatas, **kwargs)
 
+    def get_by_ids(self, ids: Sequence[str], /) -> List[Document]:
+        """Get documents by their IDs from the vectorstore.
+        Args:
+            ids: List of IDs to retrieve.
+        Returns:
+            List of Documents
+        """
+
+        documents = []
+
+        if ids is None or len(ids) == 0:
+            logging.info(EMPTY_IDS_ERROR_MESSAGE)
+        else:
+            result = self._get_documents_by_ids(ids)
+            for item in result:
+                documents.append(
+                    Document(
+                        id=item.custom_id,
```

Comment:
how can we ensure item has a custom_id, content or metadata if its not typed?

Reply:
`item` is a result from the DB where we've selected those specific columns: `custom_id`, `content`, `metadata`.
If `item` itself is not None, then we will be getting those columns. Hence why we are able to use though it's not typed.

For validation on if e.g `item.custom_id` results to `None`, the `Document` class does that validation.

I'll update the code to check that `item` is not `None`.